src/track/track.cpp

```diff
@@ -328,18 +325,34 @@ bool Track::setBeatsWhileLocked(mixxx::BeatsPointer pBeats) {
     return true;
 }
 
-void Track::setBeatsMarkDirtyAndUnlock(QMutexLocker* pLock, mixxx::BeatsPointer pBeats) {
+bool Track::trySetBeatsMarkDirtyAndUnlock(
+        QMutexLocker* pLock,
+        mixxx::BeatsPointer pBeats,
+        SetBeatsMode mode) {
```

Comment:
How about a parameter `std::option<bool> bpmLocked = std::nullopt` with default value that leaves the bpmLocked flag unchanged?

Reply:
I prefer a describtive enum here, because than you can see the intend at the caller instead of a null parameter.